### PR TITLE
Bug 1781948: Do not wrap errors from syncIngressControllerStatus

### PR DESCRIPTION
**Pull request by Miciah**
An error value from `syncIngressControllerStatus` may represent a failure to update status, in which case `syncIngressControllerStatus` itself adds a clear error message and wrapping it is therefore unnecessary; or the error value may represent a retryable error, in which case wrapping the error value loses the retryable error type and adds an inaccurate error message.

* `pkg/operator/controller/ingress/controller.go` (`ensureIngressController`): Do not wrap the error value from `syncIngressControllerStatus` when adding it to the list of returned error values.
* `pkg/util/retryableerror/retryableerror_test.go` (`TestRetryableError`): Add a test case with a nil error value, and add a nil error value to the "only retryable errors" test case to verify that nil error values are ignored.

### CODE REVIEW

File: pkg/operator/controller/ingress/controller.go [hunk 1]
```diff
@@ -611,7 +611,11 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 	}
 
 	if err := r.syncIngressControllerStatus(ci, deployment, lbService, operandEvents.Items, wildcardRecord, dnsConfig); err != nil {
-		errs = append(errs, fmt.Errorf("failed to sync ingresscontroller status: %v", err))
+		syncErr := fmt.Errorf("failed to sync ingresscontroller status: %v", err)
```
Comment (ironcladlou) [inline pkg/operator/controller/ingress/controller.go]: Would using error wrapping remove the need to put this sort of code everywhere?

Comment (ironcladlou) [reply pkg/operator/controller/ingress/controller.go]: Because then we could unwrap the errors once on the outside and check to see if any of the errors in the chain are retryable

Comment (Miciah) [reply pkg/operator/controller/ingress/controller.go]: I think so, but I am going to change this PR not to wrap `err` at all because `syncIngressControllerStatus` decorates its errors sufficient, and we don't want to log "failed to sync ingresscontroller status" if the error is a retryable error (in which case there was no failure to update status).

**Comment by Miciah** [comment]
failed to generate steps from config: step "e2e-aws-upgrade" needs a lease but no lease client provided

/retest
